src/transaction/log_recovery.c

```diff
@@ -1577,6 +1580,64 @@ log_rv_analysis_complete (THREAD_ENTRY * thread_p, int tran_id, LOG_LSA * log_ls
   return NO_ERROR;
 }
 
+/*
+ * log_rv_analysis_ha_server_state -
+ *
+ * return: error code
+ *
+ *   log_lsa(in/out):
+ *   log_page_p(in/out):
+ *   prev_lsa(in/out):
+ *   is_media_crash(in):
+ *   stop_at(in):
+ *   did_incom_recovery(in/out):
+ *
+ * Note:
+ */
+static int
+log_rv_analysis_ha_server_state (THREAD_ENTRY * thread_p, LOG_LSA * log_lsa, LOG_PAGE * log_page_p, LOG_LSA * prev_lsa,
+				 bool is_media_crash, time_t * stop_at, bool * did_incom_recovery)
+{
+  LOG_REC_HA_SERVER_STATE *ha_server_state;
+  time_t last_at_time;
+  LOG_LSA record_header_lsa;
+
+  if (is_media_crash != true)
+    {
+      goto end;
+    }
+
+  LSA_COPY (&record_header_lsa, log_lsa);
+
+  /*
+   * Need to read the donetime record to find out if we need to stop
+   * the recovery at this point.
+   */
+  LOG_READ_ADD_ALIGN (thread_p, sizeof (LOG_RECORD_HEADER), log_lsa, log_page_p);
+  LOG_READ_ADVANCE_WHEN_DOESNT_FIT (thread_p, sizeof (LOG_REC_HA_SERVER_STATE), log_lsa, log_page_p);
+
+  ha_server_state = (LOG_REC_HA_SERVER_STATE *) ((char *) log_page_p->area + log_lsa->offset);
+  last_at_time = (time_t) ha_server_state->at_time;
+
+  if (stop_at != NULL && *stop_at != (time_t) (-1) && difftime (*stop_at, last_at_time) < 0)
+    {
+      /*
+       * Reset the log active and stop the recovery process at this
+       * point. Before reseting the log, make sure that we are not
+       * holding a page.
+       */
+      log_lsa->pageid = NULL_PAGEID;
+      log_recovery_resetlog (thread_p, &record_header_lsa, prev_lsa);
+      *did_incom_recovery = true;
+
+      return NO_ERROR;
+    }
```

Comment:
```suggestion
  if (stop_at == NULL || *stop_at == (time_t)(-1) || difftime(*stop_at, last_at_time) >= 0)
    {
      goto end;
    }

  log_lsa->pageid = NULL_PAGEID;
  log_recovery_resetlog(thread_p, &record_header_lsa, prev_lsa);
  *did_incom_recovery = true;
```

NIT: 가독성 측면에서 좋아보이는 것 같아 제안드립니다.

Reply:
명환 님 피드백 반영을 위해 새로운 로그 타입이 추가될 예정입니다. 해당 작업에서 검토해 보겠습니다.

Reply:
유사 패턴 코드라 그대로 두겠습니다.